docs/build/wasm/interact/interact.md

```diff
@@ -1,5 +1,139 @@
+
+    // Use the api
+    // For example:
+    console.log((await api.rpc.system.properties()).toHuman());
+
+    process.exit(0);
+}
+
+main()
+```
+
+#### Initialise Contract Class
+
+```js
+import { Abi, ContractPromise } from '@polkadot/api-contract'
+
+// After compiling the contract a ABI json is created in the artifacts. Import the ABI:
+// If you didn't use swanky the name will be metadata.json
+import ABI from './artifacts/lottery.json'
+
+const abi = new Abi(ABI, api.registry.getChainProperties())
+
+// Initialise the contract class
+const contract = new ContractPromise(api, abi, address) // address is the deployed contract address
+```
+
+#### Query Contract Messages
+
+```js
+
+// Get the gas WeightV2 using api.consts.system.blockWeights['maxBlock']
+const gasLimit = api.registry.createType(
+    'WeightV2',
+    api.consts.system.blockWeights['maxBlock']
+  )
+
+// Query the contract message
+const { gasRequired, result, output } = await contract.query.pot(
+    account.address,
+    {
+      gasLimit,
+    }
+  )
+```
+
+#### Send Contract Transaction
+
+Sending contract transaction normally is a 2 step process. First is to query the transaction and check for errors. We have a helper function to do this for you. The second step is to send the transaction. The helper function will return the transaction object which you can use to sign and send the transaction.
+The parameters for the helper function are:
+
+- api: The api instance
+- contract: The contract instance
+- message: The message to send
+- sender: The sender address
+- value: The value to send with the transaction
+- ...params: The parameters for the message
+
+```js
+
+import { sendTransaction } from '@astar-network/astar-sdk-core';
+
+try {
+  const result = await sendTransaction(api, contract, 'enter', account.address, new BN('1000000000000000000'))
+
+  result.signAndSend(account.address, (res) => {
+    if (res.status.isInBlock) {
+      console.log('in a block')
+    }
+    if (res.status.isFinalized) {
+      console.log('finalized')
+      console.log('Successfully entered in lottery!')
+    }
+  })
+} catch (error) {
+  if (error.isErr) {
+    let display = ''
+    if (error.asErr.isModule) {
+      const dispatchError = api.registry.findMetaError(error.asErr.asModule)
+      console.log('error', dispatchError.name)
+      display = dispatchError.docs.length ? dispatchError.docs.concat().toString() : dispatchError.name
+    } else {
+      display = error.asErr.toString()
+    }
+
+    console.log(display)
+    return
+  }
+
+  if (error.isOk) {
+    const flags = error.asOk.flags.toHuman()
+    if (flags.includes('Revert')) {
+      console.log('Revert')
+      const type = contract.abi.messages[5].returnType
+      const typeName = type?.lookupName || type?.type || ''
+      const error = contract.abi.registry.createTypeUnsafe(typeName, [error.asOk.data]).toHuman()
+
+      console.log(error ? (error as any).Err : 'Revert')
+      return
+    }
+  }
+}
+```
```

Comment:
Since this code is needed for every contract call, can you make whole try/catch segment as a helper function

Reply:
Moved error handling to helper https://github.com/AstarNetwork/astar.js/pull/42